BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -618,8 +638,38 @@ public Vector3d UpdateGPSTarget()
 
         protected void UpdateHeatTarget()
         {
-
-            if (lockFailTimer > 1)
+            if (hasDataLink)
+            {
+                if(lockFailTimer > radarTimeout)
+                {
+                    targetVessel = null;
+                    TargetAcquired = false;
+                    predictedHeatTarget.exists = false;
+                    predictedHeatTarget.signalStrength = 0;
+                    return;
+                }
+                else
+                {
+                    if (!heatTarget.exists) _lockTimer = 0;
+                    if (vrd)
+                    {
+                        TargetSignatureData t = TargetSignatureData.noTarget;
+                        if (!predictedHeatTarget.exists) predictedHeatTarget = TargetSignatureData.noTarget;
+                        if (vrd.locked) t = vrd.lockedTargetData.targetData;
+                        if (t.exists)
+                        {
+                            TargetAcquired = true;
+                            predictedHeatTarget = t;
+                            predictedHeatTarget.timeAcquired = Time.time;
+                            if (!heatTarget.exists) predictedHeatTarget.signalStrength = 1;
```

Comment:
if (!heatTarget.exists) will always return true here, as your new code will only get called when !heatTarget.exists.
I also have to question what this adds that doing a radar -> heat terminal guidance doesn't? You've essentially turned the heater into a radar missile here.

Reply:
in true this code gets called if(vrd), the if (!heatTarget.exists) only resets the locktimer, this makes a heatmissile that is a full heatmissile and don't need a radar but can be guided via radar until it can get a strong lock(lock the target for more than 5 seconds) and them it disable datalink and is back to be a full normal heat seeker where a radar -> heat terminal guidance needs mandatorily a radar lock to launch it and then turn into heat